pkg/auth/authproviders/oidc/extract_claims_test.go

```diff
@@ -0,0 +1,64 @@
+package oidc
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/auth/tokens"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockClaimExtractor struct {
+	claims map[string]interface{}
+}
+
+func (e *mockClaimExtractor) Claims(input interface{}) error {
+	switch u := input.(type) {
+	case *map[string]interface{}:
+		for k, v := range e.claims {
+			(*u)[k] = v
+		}
+		return nil
+	default:
+		return errors.Errorf("unsupported type %T", input)
+	}
+}
+
+func TestExtractCustomClaims(t *testing.T) {
+	claim := &tokens.ExternalUserClaim{
+		Attributes: make(map[string][]string, 0),
+	}
+	rolesList := []interface{}{
+		"a",
+		"b",
+		"c",
+	}
+	claimExtractor := &mockClaimExtractor{
+		claims: map[string]interface{}{
+			"realm_access": map[string]interface{}{
+				"roles": rolesList,
+			},
+			"a": map[string]interface{}{
+				"b": "a-b-value",
+			},
+			"is_internal":       true,
+			"is_email_verified": false,
```

Comment:
I would suggest adding test cases for some error situations:
- a case for `Unsupported claim type` - i.e. map is not supported. Probably path `a` would cover that.
- a case for `expected next value to be of map type`. I think path `a.b.c` would cover that.